cai_causal_graph/graph_components.py

```diff
@@ -442,6 +498,33 @@ def details(self) -> str:
         """Return a detailed string description of the object."""
         return self.__repr__()
 
+    @classmethod
+    def from_dict(cls, edge_dict: dict) -> Edge:
+        """
+        Deserialize a dictionary to a `cai_causal_graph.graph_components.Edge` instance.
+
+        :param edge_dict: The dictionary representation of the `cai_causal_graph.graph_components.Edge` instance.
+        :return: The `cai_causal_graph.graph_components.Edge` instance.
+        """
+        source_node_class = edge_dict['source'].get('node_class', 'Node')
+        destination_node_class = edge_dict['destination'].get('node_class', 'Node')
+        assert source_node_class in NodeClassDict, f'Source node class not in NodeClassDict. Got {source_node_class}.'
```

Comment:
User does not know what NodeClassDict is. Suggest: "Source node class is not a valid class. Got {source_node_class} and supported types are {list(NodeClassDict)}."